File: src/Scoreboard.Domain/PlayerScore.cs
namespace Scoreboard.Domain;

public readonly record struct PlayerScore
{
    public PlayerScore(string playerName, int score)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(playerName);

        if (score < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(score), "Score must be greater than or equal to zero.");
        }

        PlayerName = playerName.Trim();
        Score = score;
    }

    public string PlayerName { get; }

    public int Score { get; }
}

File: src/com/scoreboard/domain/PlayerScore.java
package com.scoreboard.domain;

public record PlayerScore(String playerName, int score) {

  public PlayerScore {
    if (playerName == null || playerName.isBlank()) {
      throw new IllegalArgumentException("Player name must not be blank.");
    }

    if (score < 0) {
      throw new IllegalArgumentException("Score must be greater than or equal to zero.");
    }

    playerName = playerName.trim();
  }
}

